src/test/ssr.test.js

```diff
@@ -71,6 +74,37 @@ describe('ssr', () => {
     `))
   })
 
+  it('should add a nonce to the stylesheet if webpack nonce is detected in the global scope', () => {
+    injectGlobal`
+      body { background: papayawhip; }
+    `
+    const Heading = styled.h1`
+      color: red;
+    `
+
+    // eslint-disable-next-line no-underscore-dangle
+    global.__webpack_nonce__ = 'foo'
+
+    const sheet = new ServerStyleSheet()
+    const html = renderToString(sheet.collectStyles(<Heading>Hello SSR!</Heading>))
+    const css = format(sheet.getStyleTags())
+
+    expect(html).toEqual('<h1 class="sc-a b" data-reactroot="" data-reactid="1" data-react-checksum="197727696">Hello SSR!</h1>')
+    expect(css).toEqual(format(`
```

Comment:
Will do. I added some new tests for previously uncovered code which use snaps but I'll do a separate PR to convert existing tests where it makes sense.